examples/transcription/websocket/callback_stream.rs

```diff
@@ -0,0 +1,53 @@
+use std::env;
+use std::time::Duration;
+
+use futures::stream::StreamExt;
+
+use deepgram::{
+    common::options::{Encoding, Endpointing, Language, Options},
+    Deepgram, DeepgramError,
+};
+
+static PATH_TO_FILE: &str = "examples/audio/bueller.wav";
+static AUDIO_CHUNK_SIZE: usize = 3174;
+static FRAME_DELAY: Duration = Duration::from_millis(16);
+
+#[tokio::main]
+async fn main() -> Result<(), DeepgramError> {
+    let deepgram_api_key =
+        env::var("DEEPGRAM_API_KEY").expect("DEEPGRAM_API_KEY environmental variable");
+
+    let dg_client = Deepgram::new(&deepgram_api_key)?;
+
+    let options = Options::builder()
+        .smart_format(true)
+        .language(Language::en_US)
+        .build();
+
+    let callback_url = env::var("DEEPGRAM_CALLBACK_URL")
+        .expect("DEEPGRAM_CALLBACK_URL environmental variable")
+        .parse()
+        .expect("DEEPGRAM_CALLBACK_URL not a valid URL");
+
+    let mut results = dg_client
+        .transcription()
+        .stream_request_with_options(options)
+        .keep_alive()
+        .encoding(Encoding::Linear16)
+        .sample_rate(44100)
+        .channels(2)
+        .endpointing(Endpointing::CustomDurationMs(300))
+        .interim_results(true)
+        .utterance_end_ms(1000)
+        .vad_events(true)
+        .no_delay(true)
+        .callback(callback_url)
```

Comment:
Remind me, do streaming requests still return the same kinds of messages to the original caller when a callback is provided?

Reply:
Some, but not all. I thought that it returned a request ID, but I think it may come back in the headers of the HTTP upgrade. I want to be able to return this, so need to play with this PR more.

The websocket itself actually sends `SpeechSegmentStarted` back to the original caller, everything else to the callback. Which makes sense, in most cases if you have a websocket open, you want to know when the user starts from the calling code so you can handle barge-in and stuff like that.

Reply:
Maybe we can drop a request_id into the WebsocketHandle, and provide a `request_id()` method on all the interface types. (`WebsocketHandle` and `RequestStream`).

Reply:
@jcdyer I implemented a way to pull a request ID from a websocket - see the updated examples and the `TranscriptionStream::request_id` method.

We could eventually extend this to batch STT and TTS as well, but going to keep the scope limited to what is needed for the client, and focus on websockets for now.